chat_client/src/chat_receiver.py

```diff
@@ -0,0 +1,57 @@
+import logging
+import threading
+from typing import Iterator
+
+import grpc
+import protobufs.chat_pb2 as chat_pb2
+
+
+class ChatReceiver(threading.Thread):
+    """Thread class with a stop() method. The thread itself has to check
+    regularly for the stopped() condition."""
+
+    def __init__(self, response_iterator: Iterator[chat_pb2.RecieveMessagesReply]):
+        super(ChatReceiver, self).__init__()
+        self._stop_event = threading.Event()
+        self._unauth_event = threading.Event()
+        self._response_iterator = response_iterator
+        logging.basicConfig(format='%(message)s', level=logging.DEBUG)
+
+    def run(self) -> None: 
+        logging.debug("Stream started on another thread...")
+        while not self.is_stopped():
+            try:
+                response = next(self._response_iterator)
+            except grpc.RpcError as rpc_error:
+                if rpc_error.code() == grpc.StatusCode.UNAUTHENTICATED:
+                    logging.debug("User not registred...")
+                    self._unauth_event.set()
+                    return
+                elif rpc_error.code() == grpc.StatusCode.CANCELLED:
+                    logging.debug("Stream canceled by server...")
+                    self.s_stop()
```

Comment:
You may put `return` after if-else statement to not duplicate it

Reply:
> You may put `return` after if-else statement to not duplicate it

I changed this block a bit, there should be less returns right now.